stream-core/src/main/java/io/getstream/client/model/activities/BaseActivity.java

```diff
@@ -50,27 +52,33 @@
     protected String target;
     protected Date time;
 
+    public BaseActivity() {
+        to = Arrays.asList();
+    }
+
     @JsonDeserialize(contentUsing = ActivitySignedRecipientDeserializer.class)
     protected List<String> to;
 
     protected String origin;
     protected String duration;
-
-    @JsonProperty("foreign_id")
     protected String foreignId;
 
+    @JsonIgnore
     public String getId() {
         return id;
     }
 
+    @JsonProperty("id")
     public void setId(String id) {
         this.id = id;
     }
 
+    @JsonIgnore
     public String getOrigin() {
         return origin;
     }
 
+    @JsonProperty("origin")
```

Comment:
that's the only way I could find to get jackson to do "asymmetrical" de/serialisation (origin field should never be sent but should be deserialised from responses) same applies for id and duration props